ios/DomainLayer/SharedDomain/Sources/SharedDomain/Common/Extensions/ErrorResult+Extensions.swift

```diff
@@ -1,12 +1,33 @@
 import DevstackKmpShared
 
-public extension ErrorResult {
+public struct KmmLocalizedError: Swift.Error {
+    public let errorResult: ErrorResult?
+    /// Returns localized message based on shared error string
+    public let localizedMessage: String
     
-    var asError: Swift.Error {
-        switch self {
-        // CommonError
-        case is DevstackKmpShared.CommonError.NoNetworkConnection: CommonError.noNetworkConnection
-        default: CommonError.unknownError
+    public init(errorResult: ErrorResult?, localizedMessage: String) {
+        self.errorResult = errorResult
+        self.localizedMessage = localizedMessage
+    }
+}
+
+public extension Swift.Error {
+    var kmmErrorResult: ErrorResult? {
+        switch self{
+        case let error as KmmLocalizedError:
+            return error.errorResult
+        default:
+            return nil
+        }
+    }
+    
+    /// Returns localized message based on shared error string. If Error is not KmmLocalizedError - it will returns localizedDescription as usual
+    var localizedMessage: String {
```

Comment:
is this actually used somewhere? 🤔

Reply:
I think that If we want to show this on iMeetup, we need to have some iOS usage

Reply:
I fixed it, now its used in Login and Registration view models